Header head
constraint[] obsts_cst